include/RAJA/index/ListSegment.hpp

```diff
@@ -148,14 +148,30 @@ class TypedListSegment
   ///
   /// \brief Construct list segment from given array with specified length.
   ///
-  /// By default the ctor performs deep copy of array elements.
+  /// By default the ctor on the host performs deep copy of array elements.
+  /// By default the ctor on the device does not perform a copy of array of elements. 
+  ///
+  RAJA_HOST_DEVICE TypedListSegment(const value_type* values,
+                                    Index_type length) 
+    : m_data{const_cast<value_type*>(values)}, m_size{length}, m_owned{Unowned}
+  {
+    
+#if !defined(RAJA_DEVICE_CODE) && defined(RAJA_ENABLE_CUDA)
```

Comment:
@trws , @davidbeckingsale , @MrBurmark , I think this was the missing piece. I have this working working now with Tiling. The data is un-owned when compiling on the CPU.

Reply:
But its owned when CUDA is enabled, I think there is a corner case that is missing  to get it to work correctly.